src/server/game/Scripting/ScriptReloadMgr.cpp

```diff
@@ -87,11 +90,13 @@ static char const* GetSharedLibraryPrefix()
 #endif
 }
 
-// Returns "dll" on Windows and "so" on posix.
+// Returns "dll" on Windows, "dylib" on OS X, and "so" on posix.
 static char const* GetSharedLibraryExtension()
 {
 #ifdef _WIN32
     return "dll";
+#elif __APPLE__
+    return "dylib";
```

Comment:
You are also required to add an entry in the CMake function [here](https://github.com/TrinityCore/TrinityCore/blob/86b98686a95e23247ecb774fb23ecd5b8d94b97b/cmake/macros/ConfigureScripts.cmake#L89) so the shared library gets removed properly on uninstall.